### PR TITLE
docs: add content describing diff between datahub and acryl datahub

### PR DESCRIPTION
**Pull request by shirshanka**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: docs/managed-datahub/managed-datahub-overview.md [hunk 1]
```diff
@@ -1,8 +1,134 @@
...
+We’ve organized this comparison using a few categories:
+* Search and Discovery: the ability to discover data assets at your organization and understand relationships between them
+* Data Governance: features that help you govern the crown jewels of your organization, and trim out the datasets that seem to grow like weeds when no one’s looking
+* Data Quality: features that help you ensure your data pipelines are producing high quality assets, and if they’re not, you’re the first to know
+* Enterprise Grade: features needed to roll out at scale to large enterprises
+* Implementation and Support: features related to ease of deployment and maintenance
+
+## How to read this list
+
+Not a fan of checkboxes? Neither are we.
```
Comment (maggiehays) [inline docs/managed-datahub/managed-datahub-overview.md:17]: Totally agree that checkboxes would be bland here, but the current styling is pretty tough to read. At a glance, they look like previously-visited hyperlinks; it's also really hard to see in dark mode: 
CleanShot 2024-04-16 at 17 53 43

Comment (maggiehays) [reply docs/managed-datahub/managed-datahub-overview.md:17]: Also, while scrolling down, it's hard to keep the mental model of which color maps to which - by the time I got here, I couldn't easily synthesize which was DataHub vs. Acryl DataHub: 
CleanShot 2024-04-16 at 17 53 54

Comment (yoonhyejin) [reply docs/managed-datahub/managed-datahub-overview.md:17]: Agree with this - I would rather make this into markdown table with checkboxes, something like this ; https://datahubproject.io/docs/api/datahub-apis#datahub-api-comparison

Comment (yoonhyejin) [reply docs/managed-datahub/managed-datahub-overview.md:17]: Also, i think we wouldnt need "How to read this list"  if we go with the checkboxes. @shirshanka let me know if you need help with this.

File: docs/managed-datahub/managed-datahub-overview.md [hunk 2]
```diff
@@ -1,8 +1,134 @@
...
 
-Acryl DataHub offers a slew of additional features on top of the normal OSS project.
+DataHub is the #1 open source metadata platform for developers.
+Acryl DataHub takes DataHub to the next level by offering features that allow
+you to roll out the product to the entire organization beyond your central data
+platform team.
 
-## Chrome Extension
+We’ve organized this comparison using a few categories:
+* Search and Discovery: the ability to discover data assets at your organization and understand relationships between them
```
Comment (yoonhyejin) [inline docs/managed-datahub/managed-datahub-overview.md:9]: I think it would be better if we move this as a description under each title

**Comment by shirshanka** [comment]
Addressed review comments. PTAL @yoonhyejin

File: docs/managed-datahub/managed-datahub-overview.md [hunk 3]
```diff
@@ -1,8 +1,108 @@
...
+| Shared audit logs | ❌ | ✅ |
+
+
+## Implementation and Support
+Features related to ease of deployment and maintenance.
+
+| Feature | DataHub | Acryl DataHub |
+| ---------------------------------------------- | ------- | ------------- |
+| Community support | ✅ | ✅ |
+| Your own engineering team | ✅ | ❌ (they can instead focus on high-value work like contributing features to the open source product, or build amazing data applications with the APIs!)|
```
Comment (yoonhyejin) [inline docs/managed-datahub/managed-datahub-overview.md:80]:
```suggestion
| Your own engineering team | ✅ | ❌ (They can instead focus on high-value work like contributing features to the open source product, or build amazing data applications with the APIs!)|
```

File: docs/managed-datahub/managed-datahub-overview.md [hunk 4]
```diff
@@ -1,8 +1,108 @@
...
+| Your private fork of DataHub | ✅ | ❌ (you won't need to manage and maintain your own fork, upgrade to latest releases etc.)|
```
Comment (yoonhyejin) [inline docs/managed-datahub/managed-datahub-overview.md:81]:
```suggestion
| Your private fork of DataHub | ✅ | ❌ (You won't need to manage and maintain your own fork, upgrade to latest releases etc.)|
```

**Pull request merged**